core/API/DataTablePostProcessor.php

```diff
@@ -191,8 +191,8 @@ public function applyFlattener($dataTable)
     public function applyTotalsCalculator($dataTable)
     {
         if (1 == Common::getRequestVar('totals', '1', 'integer', $this->request)) {
-            $reportTotalsCalculator = new ReportTotalsCalculator($this->apiModule, $this->apiMethod, $this->request);
-            $dataTable     = $reportTotalsCalculator->calculate($dataTable);
+            $calculator = new ReportTotalsCalculator($this->apiModule, $this->apiMethod, $this->request, $this->report);
```

Comment:
By passing the report to the totals calculator we no longer have to find the report over and over again for each dataTable